modules/andes-core/broker/src/main/java/org/wso2/andes/store/file/FileMessageStoreImpl.java

```diff
@@ -0,0 +1,1364 @@
+
+        while (mutableLongIterator.hasNext()) {
+            long messageID = mutableLongIterator.next();
+
+            List<AndesMessagePart> messageContentList = new ArrayList<>();
+
+            DBIterator keyIterator = brokerStore.iterator();
+            String head = generateKey(MESSAGE_CONTENT, MESSAGE, Long.toString(messageID), "0");
+            keyIterator.seek(bytes(head));
+
+            try {
+                while (keyIterator.hasNext()) {
+                    //key = MESSAGE.$message_id.$offset.MESSAGE_CONTENT
+                    String key = asString(keyIterator.peekNext().getKey());
+                    String[] keySplit = key.split(CONNECTOR);
+                    Long currentID = Long.parseLong(keySplit[1]);
+                    String identifier = keySplit[keySplit.length - 1];
+
+                    if (currentID != messageID || !identifier.equals(MESSAGE_CONTENT)) {
+                        break;
+                    }
+                    int offset = Integer.parseInt(keySplit[2]);
+
+                    AndesMessagePart messagePart = getContent(messageID, offset);
+                    messageContentList.add(messagePart);
+
+                    keyIterator.next();
+                }
+            } finally {
+                try {
+                    keyIterator.close();
+                } catch (IOException e) {
+                    log.error("Key iterator closing failed.", e);
+                }
+            }
+            messages.put(messageID, messageContentList);
+        }
+        return messages;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AndesMessageMetadata getMetadata(long messageId) throws AndesException {
+        String messageMetaDataIdentifier = generateKey(MESSAGE_METADATA, MESSAGE, Long.toString(messageId));
+        byte[] byteMetaData = brokerStore.get(bytes(messageMetaDataIdentifier));
+        return new AndesMessageMetadata(messageId, byteMetaData, true);
+    }
+
+    /**
+     * read  a metadata list from store specifying a starting message id and a limit
+     *
+     * @param storageQueueName name of the queue messages are stored
+     * @param firstMsgId       first id of the range
+     * @param limit            amount of messages which should be returned in the list
+     * @return list of metadata
+     * @throws AndesException
+     */
+    @Override
+    public List<DeliverableAndesMetadata> getMetadataList(String storageQueueName, long firstMsgId, long limit)
+            throws AndesException {
+        List<DeliverableAndesMetadata> metadataList = new ArrayList<>();
+
+        DBIterator keyIterator = brokerStore.iterator();
+        String head = generateKey(MESSAGE_METADATA, DESTINATION, storageQueueName, Long.toString(firstMsgId));
+        keyIterator.seek(bytes(head));
+
+        try {
+            while (keyIterator.hasNext()) {
+                //key = DESTINATION.$destination_name.$message_id.MESSAGE_METADATA
```

Comment:
Commented code can be removed?

Reply:
That is a comment to show the configuration of the key. Updated it to avoid the confusion.